pep-0xxx.rst

```diff
@@ -0,0 +1,797 @@
+to load certificates from HSMs. If a common interface emerges for doing this,
+this module may be updated to provide a standard constructor for this use-case
+as well.
+
+    class Certificate(metaclass=ABCMeta):
+        @abstractclassmethod
+        def from_buffer(self, buffer: bytes) -> Certificate:
+            """
+            Creates a Certificate object from a byte buffer. This byte buffer
+            may be either PEM-encoded or DER-encoded. If the buffer is PEM
+            encoded it *must* begin with the standard PEM preamble (a series of
+            dashes followed by the ASCII bytes "BEGIN CERTIFICATE" and another
+            series of dashes). In the absence of that preamble, the
+            implementation may assume that the certificate is DER-encoded
+            instead.
+            """
+
+        @abstractclassmethod
+        def from_file(self, path: Union[pathlib.Path, AnyStr]) -> Certificate:
+            """
+            Creates a Certificate object from a file on disk. This method may
+            be a convenience method that wraps ``open`` and ``from_buffer``,
+            but some TLS implementations may be able to provide more-secure or
+            faster methods of loading certificates that do not involve Python
+            code.
+            """
+
+
+Private Keys
+~~~~~~~~~~~~
+
+This module would define an abstract private key class. Much like the
+Certificate class, this class has almost no behaviour in order to give as much
+freedom as possible to the concrete implementations to treat keys carefully.
+
+This class has all the caveats of the ``Certificate`` class.
+
+    class PrivateKey(metaclass=ABCMeta):
+        @abstractclassmethod
+        def from_buffer(self,
+                        buffer: bytes,
+                        password=None: Optional[Union[Callable[[], Union[AnyStr, bytearray]], AnyStr, bytearray]) -> Certificate:
+            """
+            Creates a PrivateKey object from a byte buffer. This byte buffer
+            may be either PEM-encoded or DER-encoded. If the buffer is PEM
+            encoded it *must* begin with the standard PEM preamble (a series of
+            dashes followed by the ASCII bytes "BEGIN", the key type, and
+            another series of dashes). In the absence of that preamble, the
+            implementation may assume that the certificate is DER-encoded
+            instead.
+
+            The key may additionally be encrypted. If it is, the ``password``
+            argument can be used to decrypt the key. The ``password`` argument
+            may be a function to call to get the password for decrypting the
+            private key. It will only be called if the private key is encrypted
+            and a password is necessary. It will be called with no arguments,
+            and it should return a string, bytes, or bytearray. If the return
+            value is a string it will be encoded as UTF-8 before using it to
```

Comment:
@Lukasa As I understand, and I could be wrong, using Shift-JIS is popular in Japan and the users who use it tend to have their locales setup for it. It doesn't seem that far out of line to assume that someone in Japan, using a Shift-JIS locale might use a Japanese letter in their password. For example:

```python

PASSWORD = "学校"

assert PASSWORD.encode("utf8") == PASSWORD.encode("shift-jis")
```